cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -496,89 +496,66 @@ def __init__(
-                        name=prop.name,
-                        description=prop.description,
-                    )
-                else:
-                    type_: CognitePropertyType
-                    if issubclass(type_cls, ListablePropertyType):
-                        type_ = type_cls(is_list=prop.is_list or False)
-                    else:
-                        type_ = cast(CognitePropertyType, type_cls())
-                    container.properties[prop_name] = dm.ContainerProperty(
-                        type=type_,
-                        nullable=prop.nullable if prop.nullable is not None else True,
-                        default_value=prop.default,
-                        name=prop.name,
-                        description=prop.description,
-                    )
-
-            uniqueness_properties: dict[str, set[str]] = defaultdict(set)
-            for prop in container_properties:
-                if prop.container_property is not None:
-                    for constraint in prop.constraint or []:
-                        uniqueness_properties[constraint].add(prop.container_property)
-            for constraint_name, properties in uniqueness_properties.items():
-                container.constraints = container.constraints or {}
-                container.constraints[constraint_name] = dm.UniquenessConstraint(properties=list(properties))
+        output = DMSSchema(
+            spaces=spaces,
+            data_models=dm.DataModelApplyList([data_model]),
+            views=views,
+            containers=containers,
+            node_types=node_types,
+        )
+        if self.include_pipeline:
+            return PipelineSchema.from_dms(output, self.instance_space)
+        return output
 
-            index_properties: dict[str, set[str]] = defaultdict(set)
-            for prop in container_properties:
-                if prop.container_property is not None:
-                    for index in prop.index or []:
-                        index_properties[index].add(prop.container_property)
-            for index_name, properties in index_properties.items():
-                container.indexes = container.indexes or {}
-                container.indexes[index_name] = BTreeIndex(properties=list(properties))
+    def _create_spaces(
+        self,
+        metadata: DMSMetadata,
+        containers: dm.ContainerApplyList,
+        views: dm.ViewApplyList,
+        data_model: dm.DataModelApply,
+    ) -> dm.SpaceApplyList:
+        used_spaces = {container.space for container in containers} | {view.space for view in views}
+        if len(used_spaces) == 1:
+            # We skip the default space and only use this space for the data model
+            data_model.space = used_spaces.pop()
+            spaces = dm.SpaceApplyList([dm.SpaceApply(space=data_model.space)])
+        else:
+            spaces = dm.SpaceApplyList([metadata.as_space()] + [dm.SpaceApply(space=space) for space in used_spaces])
```

Comment:
doesn't DMSMetadata have field `space` already, why calling method?

Reply:
To get the space object